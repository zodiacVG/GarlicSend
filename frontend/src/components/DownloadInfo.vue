<template>
  <div>
    <h4 id="titleText">{{fileInfo.name}}</h4>
    <el-image id="imageReview" :src="fileInfo.data" :preview-src-list="[fileInfo.data]"
      v-show="fileInfo.type === 'image/jpeg'">
    </el-image>
    <audio :src="fileInfo.data" v-show="fileInfo.type === 'audio/mpeg'" controls></audio>
    <div id="videoShow" v-show="fileInfo.type === 'video/mp4'">
      <video id="myVideo" controls>
        <source :src="fileInfo.data" type="video/mp4">
      </video>
    </div>
    <el-card id="cardShow" shadow="hover">
    <el-row id="downloadTimes" justify="center" type="flex">
      <el-col :span=15>
        <h3 class="infoText">剩余下载次数: {{fileInfo.limitDownloadTimes}}</h3>
      </el-col>
    </el-row>
    <el-row id="dueDateShow" justify="center" type="flex">
      <el-col :span=25>
        <h3 class="infoText">
          过期时间：{{fileInfo.dueDateTemp.getFullYear()}}年{{fileInfo.dueDateTemp.getMonth()+1}}月{{fileInfo.dueDateTemp.getDate()}}
          日0点</h3>
      </el-col>
    </el-row>
    <el-row id="fileSize" justify="center" type="flex">
      <el-col :span=15>
        <h4>文件大小：{{fileInfo.size}}M</h4>
      </el-col>
    </el-row>
    <el-row justify="center" type="flex" v-show="fileInfo.fileDescription !== ''">
      <el-col :span=15>
        <p id="fileDescription">文件描述：{{fileInfo.fileDescription}}</p>
      </el-col>
    </el-row>
    <el-button @click="DownloadClick" type="success">下载 <i class="el-icon-download"></i></el-button>
    </el-card>
    <el-progress id="progressLine" :text-inside="true" :stroke-width="26" :percentage="downloadProgress" v-show="downloadProgress !== 0">
    </el-progress>
    <PDFReview id="DownloadPDFReview" :fileInfo="fileInfo" v-show="fileInfo.type === 'application/pdf'"></PDFReview>
  </div>
</template>

<script>
import PDFReview from '../components/PDFReview'

export default {
  name: 'DownloadInfo',
  components: {
    PDFReview
  },
  data () {
    return {
      date: '',
      url: '',
      srcList: []
    }
  },
  props: {
    fileInfo: {},
    downloadProgress: Number
  },
  methods: {
    DownloadClick () {
      this.$emit('DownloadFile')
    }
  }
}
</script>

<style>
  audio {
    margin-top: 20px;
  }

  .el-progress {
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 20px;
    width: 80%;
  }

  .videoReview {
    width: 100%;
  }

  .infoText {
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
  }

  .el-image {
    padding-top: 30px;
  }

  .el-button {
    margin-bottom: 20px;
  }

  #imageReview {
    width: 100px;
    height: 100px
  }

  #downloadTimes {
    height: 30px;
  }

  #dueDateShow {
    height: 30px;
  }

  #fizeSize {
    margin-bottom: 10px;
  }

  #DownloadPDFReview {
    overflow: auto;
    height: 300px;
    width: 90%;
    margin-left: auto;
    margin-right: auto;
    margin-top: 20px;
  }

  #progressLine{
    margin-top: 10px;
  }

  #titleText {
    margin-top: 0;
    margin-bottom: 0;
  }

  #fileDescription {
    margin-top: -10px;
  }

  #Infobtn {
    display: block;
    width: 100px;
  }

  #myVideo{
    margin-top: 10px;
    width: 70%;
    height: 150px;
  }

  #cardShow{
    margin-top: 10px;
    width: 80%;
    margin-left: auto;
    margin-right: auto;
    border-radius: 15px;
    background-color: rgba(168, 208, 230, 0.9);
    border: 0px;
  }

</style>
