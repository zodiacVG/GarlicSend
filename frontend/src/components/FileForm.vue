<template>
  <div id="mainContainerComppnents">
    <el-form ref="form" :model="form" label-width="140px">
      <el-form-item label="下载次数限制">
        <el-col id="numberSelect" :span=15>
          <el-input-number v-model="form.limitDownloadTimes" size="medium" :step="5" min=1 max=100 style="width:105%"></el-input-number>
        </el-col>
      </el-form-item>
      <el-form-item label="截止下载时间">
        <el-col :span=16>
          <el-date-picker id="datePicker" type="date" size="medium" placeholder="选择日期" v-model="form.dueDate" @change="pickDate">
          </el-date-picker>
        </el-col>
      </el-form-item>
      <el-form-item label="设置密码">
        <el-col :span=1>
          <el-switch v-model="hasPassword"></el-switch>
        </el-col>
      </el-form-item>
      <el-form-item label="密码" prop="pass" v-show="hasPassword">
        <el-col :span=18>
          <el-input type="password" v-model="form.password" size="medium" autocomplete="off" placeholder="最多6位密码" maxlength=6>
          </el-input>
        </el-col>
      </el-form-item>
      <el-form-item label="文件描述">
        <el-col :span=18>
          <el-input id="descriptionInput" type="textarea" autosize v-model="form.fileDescription" placeholder="选填，最多20字符"
            maxlength=20></el-input>
        </el-col>
      </el-form-item>
      <el-row :justify="center">
        <el-col>
          <el-button type="primary" @click="onSubmitFile" round id="uploadbtn">上传</el-button>
          <el-button type="info" @click="refreshPage" round>取消</el-button>
        </el-col>
      </el-row>
    </el-form>
  </div>
</template>

<script>
export default {
  data () {
    return {
      hasPassword: false,
      form: {
        limitDownloadTimes: 10,
        dueDate: new Date(new Date().getTime() + 7 * 24 * 60 * 60 * 1000), // 截止时间选择器默认为7天
        fileDescription: '', // 文件描述
        password: ''
      }
    }
  },
  methods: {
    pickDate () {
      const myDate = new Date()
      if (myDate.getTime() + 7 * 24 * 60 * 60 * 1000 < this.form.dueDate.getTime()) { // 若超过七天将日期重设为七天
        this.$alert('请将截止日期设置在7天之内', '错误', {
          type: 'warning',
          confirmButtonText: '确定',
          callback: action => {
            this.$message({
              type: 'warning',
              message: '已将截止日期重设为7天'
            })
          }
        })
        this.form.dueDate = new Date(new Date().getTime() + 7 * 24 * 60 * 60 * 1000)
      }
    },
    onSubmitFile () {
      this.$emit('FileFormSumbmit', this.form)
    },
    refreshPage () {
      location.reload()
    }
  }
}
</script>

<style>
  #datePicker{
    width:75%;
  }

  #descriptionInput{
    width: 100%;
  }

  #numberSelect{
    width: 70%;
  }
</style>
