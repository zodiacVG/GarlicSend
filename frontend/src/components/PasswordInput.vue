<template>
  <div>
    <h2>请输入密码</h2>
    <el-input v-model="PasswordInput" placeholder="输入密码" id="passwordInput" size="medium" maxlength=6 show-password>
    </el-input>
    <el-button type="success" icon="el-icon-check" circle @click="submitPassword"></el-button>
  </div>
</template>

<script>
import 'crypto-js'
const crypto = require('crypto-js')
var errorCounter = 0 // * 计算密码输入错误次数

export default {
  data () {
    return {
      PasswordInput: ''
    }
  },
  props: {
    fileInfo: {} // 传来的文件信息，包含盐值和加密后的密码
  },
  methods: {
    submitPassword: function () { // * 进行比对
      if (errorCounter >= 3) {
        this.$message.error('密码超过三次，返回首页')
        this.$router.push({ path: '/home' })
      }
      const passwordTrans = this.saltedHash(this.PasswordInput, this.fileInfo.salt)
      if (this.fileInfo.password === passwordTrans) {
        this.$message({
          message: '密码正确',
          type: 'success'
        })
        this.$emit('passwordSuccess')
      } else {
        if (errorCounter >= 3) {
          this.$message.error('密码超过三次，返回首页')
          this.$router.push({ path: '/' })
        } else {
          this.$message.error('密码错误，请重试')
          this.PasswordInput = ''
          errorCounter++
        }
      }
    },
    saltedHash: function (str, salt) {
      let hashes = str
      for (let i = 0; i < 9; i++) {
        hashes = hashes + salt
        hashes = crypto.SHA256(hashes).toString()
      }
      return hashes
    }
  }
}
</script>

<style scoped>
  h2{
    padding-top: 50px;
  }

  .el-input{
    height: 80px;
    width:150px;
    margin-right: 10px;
}
</style>
