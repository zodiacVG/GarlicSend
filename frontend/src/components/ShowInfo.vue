<template>
  <div id="ShowInfo">
    <h2 id="tokenCode">提取码</h2>
    <el-button type="primary" id="tokenCodebtn" @click="copyCode" plain>{{fileInfo.tokenCode}}</el-button>
    <h2>提取链接</h2>
    <el-button type="primary" id="tokenCodebtn" size="mini" @click="copyLink" plain>{{fileInfo.fileLink}}</el-button>
    <p id="hintDay">
      过期时间为{{fileInfo.dueDate.getFullYear()}}年{{fileInfo.dueDate.getMonth()+1}}月{{fileInfo.dueDate.getDate()}}日0点</p>
    <el-button id="backHomebtn" type="primary" @click="backHome">回到主页</el-button>
  </div>
</template>

<script>
export default {
  data () {
    return {

    }
  },
  props: {
    fileInfo: {}
  },
  methods: {
    copyCode: function () {
      const input = document.createElement('input')
      document.body.appendChild(input)
      input.setAttribute('value', this.fileInfo.tokenCode)
      input.select()
      if (document.execCommand('copy')) {
        document.execCommand('copy')
      }
      document.body.removeChild(input)

      const h = this.$createElement // * 后面用来实现弹窗提示
      this.$notify({
        title: 'GarlicSend',
        message: h('i', { style: 'color: teal' }, '复制成功')
      })
    },
    copyLink: function () {
      const input = document.createElement('input')
      document.body.appendChild(input)
      input.setAttribute('value', this.fileInfo.fileLink)
      input.select()
      if (document.execCommand('copy')) {
        document.execCommand('copy')
      }
      document.body.removeChild(input)

      const h = this.$createElement // * 后面用来实现弹窗提示
      this.$notify({
        title: 'GarlicSend',
        message: h('i', { style: 'color: teal' }, '复制成功')
      })
    },
    backHome: function () {
      location.reload()
    }

  }
}
</script>

<style scoped>
  #backHomebtn{
    margin-bottom: 10px;
    margin-top: 0px;
  }

  #hintDay{
    font-weight: 700;
  }

  #tokenCode{
    margin-top: 0px;
  }

  .el-tag{
    margin-top: 50px;
  }
</style>
