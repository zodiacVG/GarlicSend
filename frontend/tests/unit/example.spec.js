import { shallowMount, mount } from '@vue/test-utils'
import FileForm from '@/components/FileForm.vue'
import ShowInfo from '@/components/ShowInfo.vue'
import DownloadInfo from '@/components/DownloadInfo.vue'
import PasswordInput from '@/components/PasswordInput.vue'

describe('FileForm', () => {
  const wrapper = shallowMount(FileForm)
  it('completed show all the elements', () => {
    expect(wrapper.html()).toContain('<el-form-item label="下载次数限制">')
    expect(wrapper.html()).toContain('<el-form-item label="截止下载时间">')
    expect(wrapper.html()).toContain('<el-form-item label="截止下载时间">')
    expect(wrapper.html()).toContain('<el-form-item label="设置密码">')
    expect(wrapper.html()).toContain('<el-form-item label="文件描述">')
  })
  it('initialized correct functions', () => {
    expect(typeof FileForm.methods.onSubmitFile).toBe('function')
    expect(typeof FileForm.methods.refreshPage).toBe('function')
  })
  it('its functions return correct default data', () => {
    const defaultData = FileForm.data()
    expect(defaultData.hasPassword).toBe(false)
    expect(defaultData.form.limitDownloadTimes).toBe(10)
  })
  it('initialized the correct data in an istance', () => {
    expect(wrapper.vm.$data.form.limitDownloadTimes).toBe(10)
    expect(wrapper.vm.$data.hasPassword).toBe(false)
  })
  it('test refreshPage', () => {
    expect(wrapper.vm.refreshPage()).toBe(undefined)
  })
})

describe('ShowInfo', () => {
  it('initialized correct functions', () => {
    expect(typeof ShowInfo.methods.copyCode).toBe('function')
  })
})
it('sets the correct default data', () => {
  expect(typeof ShowInfo.data()).toBe('object')
})

describe('DownloadInfo', () => {
  it('sets the correct default data', () => {
    expect(typeof DownloadInfo.data()).toBe('object')
  })
  it('initialized correct DLIfunctions', () => {
    expect(typeof DownloadInfo.methods.DownloadClick).toBe('function')
  })
})

describe('PasswordInput', () => {
  const wrapper = mount(PasswordInput, {
    data () {
      return {
        PasswordInput: '123564'
      }
    }
  })
  it('renders counter html', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('set correct data', () => {
    expect(wrapper.vm.$data.PasswordInput).toBe('123564')
  })
  it('test saltedHash', () => {
    expect(wrapper.vm.saltedHash(1, 2) === 1).toBe(false)
  })
})
