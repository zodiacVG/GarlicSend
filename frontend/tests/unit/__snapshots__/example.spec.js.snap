// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput renders counter html 1`] = `
<div>
  <h2>请输入密码</h2>
  <el-input placeholder="输入密码" id="passwordInput" size="medium" maxlength="6" show-password=""></el-input>
  <el-button type="success" icon="el-icon-check" circle=""></el-button>
</div>
`;
